test/exe/envoy_static_test.sh

```diff
@@ -1,8 +1,25 @@
 #!/bin/bash
-#
 
-set -e
+if [[ `uname` == "Darwin" ]]; then
+  echo "macOS doesn't support statically linked binaries, skipping."
+  exit 0
+fi
 
-# Validate we statically link libstdc++ and libgcc.
-DYNDEPS=$(ldd source/exe/envoy-static | grep "libstdc++\|libgcc"; echo)
-[[ -z "$DYNDEPS" ]] || (echo "libstdc++ or libgcc dynamically linked: ${DYNDEPS}"; exit 1)
+# We can't rely on the exit code alone, since lld fails for statically linked binaries.
+DYNLIBS=$(ldd source/exe/envoy-static 2>&1)
+if [[ $? != 0 && ! ${DYNLIBS} =~ "not a dynamic executable" ]]; then
+  echo "${DYNLIBS}"
+  exit 1
+fi
+
+if [[ ${DYNLIBS} =~ "libc++" ]]; then
```

Comment:
The current build rule for libc++ (i.e. `--define force_libcpp=enabled`) doesn't enforce static linking of libc++, so this one might be a bit over the top... but we also don't link against libc++ on the CI (other than on macOS, which doesn't support statically linked binaries).